gammapy/data/event_list.py

```diff
@@ -555,6 +556,46 @@ def check(self, checks="all"):
         checker = EventListChecker(self)
         return checker.run(checks=checks)
 
+    def _get_coord_from_geom(self, geom):
+        """Return MapCoord of the EventList relative to an input geometry.
+
+        Parameters
+        ----------
+        geom : `~gammapy.maps.MapGeom`
+            the geom used to define the MapCoord
+        Returns
+        -------
+        mapcoord : `~gammapy.Map.MapCoord`
+        """
+        coord = dict(skycoord=self.radec)
+
+        cols = {k.upper(): v for k, v in self.table.columns.items()}
+
+        for axis in geom.axes:
+            try:
+                col = cols[axis.name.upper()]
+                coord[axis.name] = Quantity(col).to(axis.unit)
+            except KeyError:
+                raise KeyError("Column not found in event list: {!r}".format(axis.name))
+
+        return MapCoord.create(coord, coordsys="CEL")
```

Comment:
I'd prefer to just return a dict of `{"skycoord": self.radec, "energy": energy, "...": ..., }` to avoid the `MapCoord` import. The dict can be passed to `.get_by_coord()` as well.